PWGHF/TableProducer/candidateCreatorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,655 @@
+  {
+    // initialize 3-prong vertex fitter
+    o2::vertexing::DCAFitterN<3> df;
+    df.setPropagateToPCA(propagateToPCA);
+    df.setMaxR(maxR);
+    df.setMaxDZIni(maxDZIni);
+    df.setMinParamChange(minParamChange);
+    df.setMinRelChi2Change(minRelChi2Change);
+    df.setUseAbsDCA(useAbsDCA);
+    df.setWeightedFinalPCA(useWeightedFinalPCA);
+
+    // loop over triplets of track indices
+    for (const auto& rowTrackIndexXicPlus : rowsTrackIndexXicPlus) {
+      auto cascAodElement = rowTrackIndexXicPlus.cascade_as<CascadesLinked>();
+      if (!cascAodElement.has_cascData()) {
+        continue;
+      }
+      auto casc = cascAodElement.cascData_as<CascFull>();
+      auto trackCharmBachelor0 = rowTrackIndexXicPlus.prong0_as<aod::TracksWCovDca>();
+      auto trackCharmBachelor1 = rowTrackIndexXicPlus.prong1_as<aod::TracksWCovDca>();
+      auto collision = rowTrackIndexXicPlus.collision();
+
+      // preselect cascade candidates
+      if (doCascadePreselection) {
+        if (std::abs(casc.dcaXYCascToPV()) > dcaXYToPVCascadeMax) {
+          continue;
+        }
+        if (std::abs(casc.mXi() - massXiMinusFromPdg) > massToleranceCascade) {
+          continue;
+        }
+      }
+
+      //----------------------Set the magnetic field from ccdb---------------------------------------
+      /// The static instance of the propagator was already modified in the HFTrackIndexSkimCreator,
+      /// but this is not true when running on Run2 data/MC already converted into AO2Ds.
+      auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+      if (runNumber != bc.runNumber()) {
+        LOG(info) << ">>>>>>>>>>>> Current run number: " << runNumber;
+        initCCDB(bc, runNumber, ccdb, isRun2 ? ccdbPathGrp : ccdbPathGrpMag, lut, isRun2);
+        bz = o2::base::Propagator::Instance()->getNominalBz();
+        LOG(info) << ">>>>>>>>>>>> Magnetic field: " << bz;
+      }
+      df.setBz(bz);
+
+      //----------------accessing particles in the decay chain-------------
+      auto trackPionFromXi = casc.bachelor_as<aod::TracksWCovDca>(); // pion <- xi track from TracksWCovDca table
+
+      //--------------------------info of V0 and cascades track from LF-tables---------------------------
+      std::array<float, 3> vertexV0 = {casc.xlambda(), casc.ylambda(), casc.zlambda()};
+      std::array<float, 3> pVecV0 = {casc.pxlambda(), casc.pylambda(), casc.pzlambda()};
+      std::array<float, 3> vertexCasc = {casc.x(), casc.y(), casc.z()};
+      std::array<float, 3> pVecCasc = {casc.px(), casc.py(), casc.pz()};
+      std::array<float, 21> covCasc = {0.};
+
+      //----------------create cascade track------------------------------------------------------------
+      constexpr int MomInd[6] = {9, 13, 14, 18, 19, 20}; // cov matrix elements for momentum component
+      for (int i = 0; i < 6; i++) {
+        covCasc[MomInd[i]] = casc.momentumCovMat()[i];
+        covCasc[i] = casc.positionCovMat()[i];
+      }
```

Comment:
For my understanding:
 - the 1st line sets the elements with indices `9, 13, 14, 18, 19, 20`
 - the 2nd line sets the elements with indices `0, 1, 2, 3, 4, 5`
 
If I am not overlooking anything, it means that elements with indices `6, 7, 8, 10, 11, 12, 15, 16, 17, 21` are left to 0. Is it what you want?

Reply:
Yes. In the CascCovs table only the independent covariance matrices of the cascade decay vertex position and the cascade momenta are stored. The indices you mentioned are the ones that point to the covariance matrix elements containing the covariances between momenta and vertex coordinates (lower triangular form), which are not stored for the DCA fitter.
This is done analogously to the XicZero/OmegaCZero workflow.
As a remark, for the KF case, the full covariance matrix is stored.